File: python/flask/tryflask/app/services/service.py
from repositories.productRepository import ProductRepository
from flask import abort
from exceptions import *


class Service:
    def __init__(self, repository):
        self.repository = repository

    def bulk_create_service(self, payloads):
        try:
            self.repository.bulk_create(payloads)
        except Exception as e:
            raise InternalServerErrorException(e)

    def create_service(self, payload):
        try:
            self.repository.create(payload)
        except Exception as e:
            raise InternalServerErrorException(e)

    def update_service(self, payloads, id: str):
        try:
            res = self.repository.update(payloads, id)
            if not res:
                raise NotFoundException
        except Exception as e:
            raise InternalServerErrorException(e)

    def delete_service(self, id: str):
        try:
            res = self.repository.delete(id)
            if not res:
                raise NotFoundException
        except Exception as e:
            raise InternalServerErrorException(e)

    def find_by_id_service(self, id: str):
        try:
            res = self.repository.find_by_id(id)
            if not res:
                raise NotFoundException
            return res
        except Exception as e:
            raise InternalServerErrorException(e)

    def index_service(self, page: int, limit: int):
        try:
            return self.repository.index(page, limit)
        except Exception as e:
            raise InternalServerErrorException(e)

File: project/waraung_app/service/Service.go
package service

import (
	PGGormRepository "github.com/cocoasterr/gowarungapp/infra/db/postgres/repository/gorm"
)

type ServiceRepository struct {
	Repo PGGormRepository.PGGormRepository
}

func CreateNewService(repo PGGormRepository.PGGormRepository) *ServiceRepository {
	return &ServiceRepository{
		Repo: repo,
	}
}

func (s *ServiceRepository) CreateService(model interface{}) error {
	if err := s.Repo.Create(model); err != nil {
		return err
	}
	return nil
}
func (s *ServiceRepository) IndexService(tbName string, page, limit int) ([]map[interface{}]interface{}, error) {
	res, err := s.Repo.Index(tbName, page, limit)
	if err != nil {
		return nil, err
	}
	return res, nil
}

func (s *ServiceRepository) FinByService(tbName, key, value string) ([]map[interface{}]interface{}, error) {
	res, err := s.Repo.FindBy(tbName, key, value)
	if err != nil {
		return nil, err
	}
	var result []map[interface{}]interface{}
	result = append(result, res)
	return result, nil
}

func (s *ServiceRepository) UpdateService(model interface{}, id, tbName string) error {
	if err := s.Repo.Update(model, id, tbName); err != nil {
		return err
	}
	return nil
}

func (s *ServiceRepository) DeleteService(model interface{}, id string) error {
	if err := s.Repo.Delete(model, id); err != nil {
		return err
	}
	return nil
}
